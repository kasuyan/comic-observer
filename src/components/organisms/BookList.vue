<template>
  <div>
    <ul>
      <BookListItem
        v-for="(item, index) in bookData"
        :key="item.isbn"
        :data="item"
        :index="index"
        :onSelectData="onSelectData"
      />
    </ul>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import BookListItem from "../molecules/BookListItem.vue";

export interface BookData {
  isbn: number;
  name: string;
  vol: number;
  description: string;
  image: string;
  updata: boolean;
}

@Component({
  components: {
    BookListItem
  }
})
export default class BookList extends Vue {
  @Prop() private bookData!: BookData[];
  @Prop() private msg!: string;
  @Prop() onSelectData!: (index: number) => void;
}
</script>

<style scoped>
h1 {
  color: red;
}

ul {
  margin: 0 0 6rem;
  padding: 0;
}
</style>
