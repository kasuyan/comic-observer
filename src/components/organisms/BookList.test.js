import { shallowMount } from "@vue/test-utils";
import BookList from "./BookList.vue";

describe("BookList.vue", () => {
  it("matches snapshot", () => {
    const items = ["item 1", "item 2"];
    const wrapper = shallowMount(BookList, {
      bookData: items
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
