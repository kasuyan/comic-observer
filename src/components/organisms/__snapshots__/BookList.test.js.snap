// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList.vue matches snapshot 1`] = `
"<div>
  <ul></ul>
</div>"
`;
